main.tf

```diff
@@ -135,7 +135,13 @@ resource "ibm_is_instance" "vsi" {
     }
     content {
       subnet = network_interfaces.value.id
-      security_groups = flatten([
+      security_groups = length(flatten([
```

Comment:
@ocofaigh any ideas how I could avoid code duplication here?

Reply:
@jor2 Its pretty hard to follow that logic - could you maybe add some code comments to explain what its doing?

Reply:
Added some comments. It just checks if security group list is empty don't pass empty list, instead pass default security group. This avoid reapply failures.

Reply:
ok thanks, comment is good. As for code duplication, have you tried using locals?